xdsl/transforms/lower_riscv_func.py

```diff
@@ -0,0 +1,93 @@
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.ir import MLContext, Operation
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriteWalker,
+    op_type_rewrite_pattern,
+    RewritePattern,
+    PatternRewriter,
+)
+from xdsl.dialects import riscv, riscv_func, riscv_structured
+from xdsl.transforms.dead_code_elimination import dce
+
+SCALL_EXIT = 93
+"""
+93 is the number of the `exit` syscall on RISCV.
+"""
+
+
+class LowerRISCVFuncOp(RewritePattern):
+    """
+    Temporary lowering of only the main function, would like to lower non-main functions
+    soon.
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: riscv_func.FuncOp, rewriter: PatternRewriter):
+        body = op.func_body.block
+        # TODO: replace with insert_at_start
```

Comment:
please address todo

Reply:
Actually I'm not sure what I meant exactly, I think we can just delete the comment